src/Rx/Weno/SetPharmacyController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c )2019. Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Rx\Weno;
+
+use OpenEMR\Common\Database\Connector;
+use OpenEMR\Entities\Pharmacies;
+
+/**
+ * Class SetPharmacyController
+ *
+ * @package OpenEMR\Rx\Weno
+ */
+class SetPharmacyController
+{
+    /**
+     * @var
+     */
+    private $entityManager;
+    private $repository;
+
+    /**
+     * SetPharmacyController constructor.
+     */
+    public function __construct()
+    {
+        $database = Connector::Instance();
+        $entityManager = $database->entityManager;
+        $this->repository = $entityManager->getRepository(Pharmacies::class);
+    }
+
+    /**
+     * @param  $term
+     * @return mixed
+     */
+    public function getPharmacyApi($term)
+    {
+        $pharmacies = $this->repository;
+        $pharmacy = $pharmacies->findAllMatch($term);
+        return  json_encode($pharmacy);
+    }
+}
+
+if (!empty($_GET['term'])) {
+    $term = filter_input(INPUT_GET, 'term', FILTER_SANITIZE_SPECIAL_CHARS);
+    $list = new SetPharmacyController();
+    echo $rlist = $list->getPharmacyApi($term);
+} else {
+    echo json_encode(["error" => "No term sent!"]);
+}
```

Comment:
This block of code can't exist with the class script. Let me know if you need guidance how to migrate this out of this class script.

Reply:
The class script ends at line 34. The class is being called from outside by an ajax request from the browser. The only purpose of this class is to set the patient's pharmacy. So, the logic follows the ajax get request sends the pharmacy id to the set pharmacy controller and the class sets the id in the patient table.
The PatientDataRepository class is where the database query is called.

Reply:
I can see where you would say to set the controller class in a file separated from the get request. You could do it that way. But why create an extra file when it is not totally necessary. The system works like it is scripted now. I know it is your final call on how the file structure should flow. So, if you want me to take the if statement and put it in another file and call the class from that file. I will do it.

Reply:
Recommend separating this into a script like prescriptionorder.php
(weird things can happen with class autoloading and placing running code in classes; it also makes the class much less flexible)